docs/proposals/status.md

```diff
@@ -0,0 +1,300 @@
+# Status
+
+## Background
+Application CRD has a list of GroupKinds and a LabelSelector to logically group live objects in the cluster. The Application CRD also provides a way to attach descriptive metadata to the logical grouping. 
+
+## Objective
+- Enhance Application CRD controller to provide status tracking for the objects that matches its selectors. 
+- Simplify the status tracking requirements for a consumer of Application CRD. 
+- Status aggregation should aim to reduce the entropy of information that the consumers of Application CRD's `.status` would otherwise have to deal with. If a consumer needs a more detailed status of the underlying components, they could query the underlying components status.
+- Propose a minimal set of standardized Conditions for all resources to implement.
+
+## Scope
+The scope is limited to the objects matching the application crd selectors. The scope is also limited to using the information present in the `.status` field for kubernetes objects. External systems to augment health (black/white box) is not part of this scope.
+
+## Proposal
+The proposal is to add a reconciler to the Application CRD controller, that creates watches for the GroupKinds that the application CRD refers to. The reconciler then inspects the matching objects and updates the aggregate status in the Application CRD. Lowering information entropy and reducing the amount of interpertation to be done by the consumers of status is important for the adoption.
+
+### Fields
+Fields are a structured way to capture the status of resources. As part of this proposal, no standardized fields are proposed. The authors opinion is it does not provide additional information than what conditions provide. If the community strongly feels about standardizing some fields for in-tree and custom resources, it can be considered. 
+
+### Conditions
+[Conditions](https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#typical-status-properties) are used to describe the status of resources in an extensible way. Out of tree controllers support a variety of conditions. There is no consistent usage across the core controller. This is exacerbated with custom controllers. To tackle this diversity and its impact in status computation for Application CRD, we need a short term and a long term solution. Short term solution is to do aggregation of diverse core resources in the Application CRD controller. Long term is to identify best practices and standardize conditions and propagate them to the core controllers thereby making Application CRD aggregation simpler.
+
+#### Standard Conditions
```

Comment:
Would be nice to have a formalized condition indicating whether this resource is awaiting GC, as well. Some resources default to bg deletion, and so on.

Reply:
Would 'Ready' condition with '.status = false'  and set to '.reason = Finalizing' be sufficient ?
We need to also consider consistency. Deletion timestamp is set by API server synchronously. It is different from the controller setting the condition (potentially delayed). System relying on this condition may miss it during the delay.

Reply:
Are the values of  `.reason` intended to be common across all controllers? If so, I think that's fine. If not, I do think it merits its own condition, as it is a fundamental notion in Kubernetes, and it will only get _more_ fundamental as time goes on. (_c.f._, proposals that rollouts of `ConfigMap` are implemented using GC, _etc._)

Reply:
.reason is not common. We Could add a finalizing condition.